operator/v1/types_network.go

```diff
@@ -562,6 +562,31 @@ type IPsecConfig struct {
 	// +kubebuilder:validation:Enum=Disabled;External;Full
 	// +optional
 	Mode IPsecMode `json:"mode,omitempty"`
+
+	// full defines configuration parameters for the IPsec `Full` mode.
+	// This will be effective only when mode is configured with `Full`,
+	// otherwise it will be ignored.
+	// +optional
+	Full IPsecFullModeConfig `json:"full,omitempty"`
```

Comment:
This needs to be a pointer, you can't have an omitempty with a non-pointer struct

Reply:
done.